cmake/setup_cached_variables.cmake

```diff
@@ -129,6 +129,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
+# Unfortunately CMake doesn't support custom configuration types so we just
+# call "DebugRelease" a "Release" build as well.
+#
+IF(CMAKE_BUILD_TYPE MATCHES "Release")
```

Comment:
Since @tjhei already fell in this trap (and I would have too), could you change this to 
```
IF(CMAKE_BUILD_TYPE MATCHES "(Debug)?Release")
```
of whatever the appropriate regex is?